### PR TITLE
[CLEANUP] Use self as the return type for fromHtml

### PR DESCRIPTION
**Pull request by oliverklee**

### CODE REVIEW

File: src/Emogrifier/HtmlProcessor/AbstractHtmlProcessor.php [hunk 1]
```diff
@@ -60,7 +60,7 @@ private function __construct()
      *
      * @throws \InvalidArgumentException if $unprocessedHtml is anything other than a non-empty string
      */
-    public static function fromHtml(string $unprocessedHtml): AbstractHtmlProcessor
+    public static function fromHtml(string $unprocessedHtml): self
```
Comment (JakeQZ) [inline src/Emogrifier/HtmlProcessor/AbstractHtmlProcessor.php]: Um, also for `::fromDomDocument` please :)

Comment (oliverklee) [reply src/Emogrifier/HtmlProcessor/AbstractHtmlProcessor.php]: Oh. Right. (Note to myself: Have coffee and breakfast first. Code after that, not before.)

Comment (oliverklee) [inline src/Emogrifier/HtmlProcessor/AbstractHtmlProcessor.php]: Note: I'd like to keep the `@return static` in the PHPDoc to make it explicit that we return an instance of the corresponding called class.

**Comment by oliverklee** [comment]
Updated and repushed.

File: src/Emogrifier/HtmlProcessor/AbstractHtmlProcessor.php (hunk 1 above)
Comment (JakeQZ) [reply src/Emogrifier/HtmlProcessor/AbstractHtmlProcessor.php]: > keep the `@return static` in the PHPDoc to make it explicit

I strongly agree.  There is still not much correlation between PHPDoc and type hints and [probably never will be](https://wiki.php.net/rfc/arrayof).

Comment (JakeQZ) [reply src/Emogrifier/HtmlProcessor/AbstractHtmlProcessor.php]: > Have coffee and breakfast first

Or as the newspaper-reading types do: look at the main article AND the side article :)

File: src/Emogrifier/HtmlProcessor/AbstractHtmlProcessor.php [hunk 2]
```diff
@@ -45,7 +45,7 @@ abstract class AbstractHtmlProcessor
     /**
      * The constructor.
      *
-     * Please use ::fromHtml instead.
+     * Please use `::fromHtml` or `::fromDomDocument` instead.
```
Comment (JakeQZ) [inline src/Emogrifier/HtmlProcessor/AbstractHtmlProcessor.php]: Good correction.  Coffee clearly working :)

**Pull request merged**